appserver/web/web-glue/src/main/java/com/sun/enterprise/web/WebComponentInvocation.java

```diff
@@ -160,4 +160,18 @@ public boolean equals(Object obj) {
             return eq;
         }
     }
+
+    @Override
+    public int hashCode() {
+        return instance == null ? container.hashCode() : container.hashCode() ^ instance.hashCode();
```

Comment:
How about using Objects.hashCode(container, instance)?

Reply:
If there are other changes requested I use it for better readability. Just this alone I feel is to small to go through a new round.

Reply:
So I am back here (see comments down there) ... 
We have two implementations of this class
* WebComponentInvocation - now you say two invocations are equal if they are done in same container and same instance. Are you sure it is alright?
* EjbInvocation - this uses Object.equals, so two invocations are equals if they is one invocation

These all and even more are events ... 
![image](https://user-images.githubusercontent.com/302904/80946476-6a195780-8dee-11ea-8f99-a3da21a2b5d9.png)

Reply:
There are even more implementations. That is part of the reason for the chosen solution.

Generally it seems reasonable for an invocation to only assume two instances being equal when `a == b` as they basically represent a call stack element. However, callers themselves only reacting to events have trouble to pass the very same instance of the invocation to both pre- and post as those originate from different execution stacks. The point where those are within one method is either too far does not even exist when originating from scheduled executions of working events.

Given this the question becomes: how do we know the pre/post invocation instance represent the same stack element? If `==` is no option `equals` is the next reasonable choice. `WebComponentInvocation` are always constructed from a `WebModule` and possibly an `instance`. Being more strict here is better since behaviour before was `==`.